### ISSUE TITLE
Add implicit keyword to the Modifiers section

### ISSUE THREAD

**Issue by fedeoasi** [scala/docs.scala-lang#1075]
The current order for modifiers is defined as:

1. Annotations
2. Override modifier
3. Access modifier
4. Final modifier
5. def

Where should the implicit keyword go? I would argue between 4) and 5) but that's not for me to decide. Is there an undocumented best practice that can be added to the docs?

**Comment by SethTisue** [comment]
I have a bunch of open source Scala repos on my laptop, because of my community build work, so I did:

```
% grep -r "final implicit" */*/src */src | wc -l
      48
% grep -r "implicit final" */*/src */src | wc -l
     442
```

this surprised me, I expected `final implicit` to win, like you did.

as a sanity check, I browsed the full results to make sure they weren't dominated by a handful of outlier repos, but they weren't.

also,

```
% grep -r "implicit private" */*/src */src | wc -l
      22
% grep -r "private implicit" */*/src */src | wc -l
     265
```
based on this survey, I would be in favor of a pull request enshrining "between 3 and 4" in the style guide.
